eval/src/vespa/eval/tensor/sparse/sparse_tensor_join.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "sparse_tensor_join.h"
+#include "sparse_tensor_t.h"
+#include "sparse_tensor_address_combiner.h"
+#include "direct_sparse_tensor_builder.h"
+
+namespace vespalib::tensor::sparse {
+
+template <typename LCT, typename RCT, typename Function>
+std::unique_ptr<Tensor>
+join(const SparseTensor &lhs_in, const SparseTensor &rhs_in, Function &&func)
+{
+    using OCT = typename eval::UnifyCellTypes<LCT,RCT>::type;
```

Comment:
if a sparse tensor can be of type double, this might not be correct in all cases (joining double with tensor)